src/thor/trace_route_action.cc

```diff
@@ -207,7 +197,8 @@ thor_worker_t::map_match(valhalla_request_t& request,
 
     // Throw exception if not trace attributes action and disconnected path.
     // TODO - perhaps also throw exception if use_timestamps and disconnected path?
-    if (request.options.action() == DirectionsOptions::trace_route && disconnected_edges.size()) {
+    if (request.options.action() == DirectionsOptions::trace_route &&
+        (disconnected_edges.size() || path_edges.empty())) {
       throw valhalla_exception_t{442};
```

Comment:
here we throw 442 but above we do `std::exception` and change it to 442. we should probably settle on one or the other, which do you prefer?

Reply:
Good catch, I think throwing a generic exception here makes sense